cpp/include/raft/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -40,6 +42,23 @@
 
 namespace raft::neighbors::cagra::detail {
 
+template <typename DataT, typename accessor>
+ivf_pq::index_params get_default_ivf_pq_build_params(
+  mdspan<const DataT, matrix_extent<int64_t>, row_major, accessor> dataset,
+  const raft::distance::DistanceType metric)
+{
+  auto build_params    = ivf_pq::index_params{};
+  build_params.n_lists = dataset.extent(0) < 4 * 2500 ? 4 : (uint32_t)(dataset.extent(0) / 2500);
+  build_params.pq_dim  = raft::Pow2<8>::roundUp(dataset.extent(1) / 2);
```

Comment:
We generally set this to something like n_cols / 4 or n_cols / 8 by default. I think compression ratio of 4 makes a reasonable default here.